File: 1926/d.py
# Python is too slow...

for _ in range(int(input())):
    n = int(input())
    a = list(map(int, input().split()))
    cnt = {}
    answer = 0
    for x in a:
        target = (2**31 - 1) ^ x
        if target in cnt and cnt[target] > 0:
            answer += 1
            cnt[target] -= 1
            if cnt[target] == 0:
                del cnt[target]
        else:
            cnt[x] = cnt.get(x, 0) + 1
    print(answer + sum(cnt.values()))

File: 1926/d.cpp
#include <iostream>
#include <map>
using namespace std;

int main()
{
  int t;
  cin >> t;
  while (t--)
  {
    int n, x;
    cin >> n;
    map<int, int> cnt;
    int answer = 0;
    for (int i = 0; i < n; ++i)
    {
      cin >> x;
      int target = (1LL << 31) - 1 ^ x;
      if (cnt.find(target) != cnt.end() && cnt[target] > 0)
      {
        answer += 1;
        cnt[target] -= 1;
        if (cnt[target] == 0)
        {
          cnt.erase(target); // Remove the entry if its count is 0 to keep the map clean
        }
      }
      else
      {
        cnt[x] += 1;
      }
    }
    int sum = 0;
    for (auto &pair : cnt)
    {
      sum += pair.second;
    }
    cout << answer + sum << endl;
  }
  return 0;
}